proxy/src/transport/tls/connection.rs

```diff
@@ -0,0 +1,78 @@
+use std::io;
+use std::net::SocketAddr;
+
+use bytes::Buf;
+use futures::Future;
+use tokio::prelude::*;
+use tokio::net::TcpStream;
+
+use transport::{AddrInfo, Io};
+
+use super::{
+    rustls::ServerSession,
+    tokio_rustls::{ServerConfigExt, TlsStream},
+
+    ServerConfig,
+};
+
+// In theory we could replace `TcpStream` with `Io`. However, it is likely that
+// in the future we'll need to do things specific to `TcpStream`, so optimize
+// for that unless/until there is some benefit to doing otherwise.
+#[derive(Debug)]
+pub struct Connection(TlsStream<TcpStream, ServerSession>);
```

Comment:
this would avoid Read/Write, but all of the other traits would have to be implemented, and since neither `TlsStream` nor the io types are defined here, i believe the wrapper type is necessary.

Reply:
Ah, yeah, you're right, nevermind.